api/provider/aws/test_aws/service_read_test.go

```diff
@@ -0,0 +1,224 @@
+	deployments := []*ecs.Deployment{
+		deployment,
+	}
+
+	loadBalancer := &ecs.LoadBalancer{}
+	loadBalancer.SetLoadBalancerName("l0-test-lb_id")
+	loadBalancers := []*ecs.LoadBalancer{
+		loadBalancer,
+	}
+
+	service := &ecs.Service{}
+	service.SetDeployments(deployments)
+	service.SetDesiredCount(int64(1))
+	service.SetLoadBalancers(loadBalancers)
+	service.SetPendingCount(int64(0))
+	service.SetRunningCount(int64(1))
+	services := []*ecs.Service{
+		service,
+	}
+
+	describeServicesOutput := &ecs.DescribeServicesOutput{}
+	describeServicesOutput.SetServices(services)
+
+	mockAWS.ECS.EXPECT().
+		DescribeServices(describeServicesInput).
+		Return(describeServicesOutput, nil)
+
+	target := provider.NewServiceProvider(mockAWS.Client(), tagStore, mockConfig)
+	result, err := target.Read("svc_id")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expectedDeployments := []models.Deployment{
+		{
+			Created:       time.Time{},
+			DeployID:      "dpl_id",
+			DeployName:    "dpl_name",
+			DeployVersion: "1",
+			DesiredCount:  1,
+			PendingCount:  0,
+			RunningCount:  1,
+			Status:        ecs.DesiredStatusRunning,
+			Updated:       time.Time{},
+		},
+	}
+
+	expected := &models.Service{
+		Deployments:      expectedDeployments,
+		DesiredCount:     1,
+		EnvironmentID:    "env_id",
+		EnvironmentName:  "env_name",
+		LoadBalancerID:   "lb_id",
+		LoadBalancerName: "lb_name",
+		PendingCount:     0,
+		RunningCount:     1,
+		ServiceID:        "svc_id",
+		ServiceName:      "svc_name",
+	}
+
+	assert.Equal(t, expected, result)
+}
+
+func TestServiceRead_serviceNotFoundError(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	tags := models.Tags{
+		{
+			EntityID:   "svc_id",
+			EntityType: "service",
+			Key:        "environment_id",
+			Value:      "env_id",
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	deployment := &ecs.Deployment{}
+	deployment.SetCreatedAt(time.Time{})
+	deployment.SetDesiredCount(int64(1))
+	deployment.SetPendingCount(int64(0))
+	deployment.SetRunningCount(int64(1))
+	deployment.SetStatus(ecs.DesiredStatusRunning)
+	deployment.SetTaskDefinition("arn:aws:ecs:region:012345678910:task-definition/dpl_id:1")
+	deployment.SetUpdatedAt(time.Time{})
+
+	deployments := []*ecs.Deployment{
+		deployment,
+	}
+
+	service := &ecs.Service{}
+	service.SetDeployments(deployments)
+	service.SetDesiredCount(int64(1))
+	service.SetPendingCount(int64(0))
+	service.SetRunningCount(int64(1))
+
+	services := []*ecs.Service{
+		service,
+	}
+
+	describeServicesOutput := &ecs.DescribeServicesOutput{}
+	describeServicesOutput.SetServices(services)
+
+	awsError := awserr.New("ServiceNotFoundException", "", nil)
+
+	mockAWS.ECS.EXPECT().
+		DescribeServices(gomock.Any()).
```

Comment:
Wouldn't we want to pass `services` as a param to this call?

Reply:
I'm actually stripping all this setup out as per zpatrick's comment at the top of the func.